sympy/geometry/line.py

```diff
@@ -999,33 +1000,19 @@ class Line(LinearEntity):
     x
     """
 
-    def __new__(cls, p1, pt=None, slope=None, **kwargs):
+    def __new__(cls, p1, *pts, **kwargs):
         if isinstance(p1, LinearEntity):
-            p1, pt = p1.args
-        else:
-            p1 = Point(p1)
-        if pt is not None and slope is None:
-            try:
-                p2 = Point(pt)
-            except NotImplementedError:
-                raise ValueError('The 2nd argument was not a valid Point. '
-                'If it was a slope, enter it with keyword "slope".')
-        elif slope is not None and pt is None:
-            slope = sympify(slope)
-            if slope.is_finite is False:
-                # when infinite slope, don't change x
-                dx = 0
-                dy = 1
-            else:
-                # go over 1 up slope
-                dx = 1
-                dy = slope
-            # XXX avoiding simplification by adding to coords directly
-            p2 = Point(p1.x + dx, p1.y + dy)
+            ambient_dim = len(p1.p1)
         else:
-            raise ValueError('A 2nd Point or keyword "slope" must be used.')
+            if iterable(p1) and not hasattr(p1, '__len__'):
```

Comment:
I did this to be efficient.  Perhaps it is confusing.  `x=[1, 2, 3]` will satisfy `is_sequence(x) == True`, but it will not make it through the if clause since `hasattr(x, '__len__') == True`.  Thus, I avoid an unnecessary call to `tuple(x)`.  Should I be doing a `try` `except` approach instead?